### PR TITLE
UCT/RDMACM: Pack dest-gid from qp_attr instead of CM id

### PR DESCRIPTION
**Pull request by evgeny-leksikov**
## What
Pack dest-gid from qp_attr instead of CM id if qp_attr.ah_attr.is_global

### CODE REVIEW

**Comment by evgeny-leksikov** [comment]
failure is https://github.com/openucx/ucx/issues/4616
bot:pipe:retest

File: src/uct/ib/rdmacm/rdmacm_cm.c [hunk 1]
```diff
@@ -201,7 +211,8 @@ static ucs_status_t uct_rdmacm_cm_id_to_dev_addr(struct rdma_cm_id *cm_id,
                                 UCT_IB_ADDRESS_PACK_FLAG_SUBNET_PREFIX;
     }
 
-    params.gid  = cm_id->route.addr.addr.ibaddr.dgid;
+    params.gid  = qp_attr.ah_attr.is_global ? qp_attr.ah_attr.grh.dgid :
+                  cm_id->route.addr.addr.ibaddr.dgid;
```
Comment (yosefe) [inline src/uct/ib/rdmacm/rdmacm_cm.c:215]: why need to check is_global?

Comment (yosefe) [reply src/uct/ib/rdmacm/rdmacm_cm.c:215]: else if (is_global)  {
  set gid for qpattr
  set pack flags UCT_IB_ADDRESS_PACK_FLAG_INTERFACE_ID..
}

File: src/uct/ib/rdmacm/rdmacm_cm.c [hunk 2]
```diff
@@ -196,12 +197,26 @@ static ucs_status_t uct_rdmacm_cm_id_to_dev_addr(struct rdma_cm_id *cm_id,
...
         params.roce_info      = roce_info;
         params.flags         |= UCT_IB_ADDRESS_PACK_FLAG_ETH;
+    } else if (qp_attr.ah_attr.is_global) {
+        params.flags         |= UCT_IB_ADDRESS_PACK_FLAG_SUBNET_PREFIX |
+                                UCT_IB_ADDRESS_PACK_FLAG_INTERFACE_ID;
     } else {
-        params.flags         |= UCT_IB_ADDRESS_PACK_FLAG_INTERFACE_ID |
-                                UCT_IB_ADDRESS_PACK_FLAG_SUBNET_PREFIX;
+        /* For local IB address, assume the remote subnet prefix is the same
+         * and pack it to make reachability check pass*/
```
Comment (yosefe) [inline src/uct/ib/rdmacm/rdmacm_cm.c:205]: minor: space after '*'

Comment (evgeny-leksikov) [reply src/uct/ib/rdmacm/rdmacm_cm.c]: fixed and squashed

**Comment by evgeny-leksikov** [comment]
@yosefe CI is passed, GH didn't update status. is this ok to merge?

**Pull request merged**